### PR TITLE
Feature/jmeter

### PR DESCRIPTION
**Pull request by lokeshrangineni**

### CODE REVIEW

**Review by rhkp — APPROVED** [review]

File: automation/ansible/roles/setup_jmeter_client/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+- name: Install jmeter client packages.
+  become: yes
+  become_user: root
+  ansible.builtin.package:
+    name: "{{ jmeter_packages }}"
+    state: present
+
+- name: Prepare JMeter client directory
+  shell: |
+    if [ ! -d "{{ jmeter_remote_bin_dir_full_path }}" ]; then
```
Comment (sgahlot) [inline automation/ansible/roles/setup_jmeter_client/tasks/main.yml:11]: You can change this to use ansible file module. This was just a easy way for me to test out my changes.

File: automation/ansible/roles/setup_jmeter_client/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      mkdir -p "{{ jmeter_remote_bin_dir_full_path }}"
+    fi
+  register: prepare_jmeter_dir_output
+
+- name: "Download jmeter binary {{ jmeter_binary_archive }}"
+  get_url:
+    url: "{{ jmeter_binary_download_uri }}/{{ jmeter_binary_archive }}"
+    dest: "{{ jmeter_remote_bin_dir_full_path }}/{{ jmeter_binary_archive }}"
+    mode: '0744'
```
Comment (sgahlot) [inline automation/ansible/roles/setup_jmeter_client/tasks/main.yml:20]: Mode should be 644 as this is a zip file

File: automation/ansible/roles/setup_jmeter_client/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    mode: 0744
+  loop:
+    - "{{ jmeter_jmx_file }}"
+    - gen_throughput.sh
+
+- name: Copy jmeter jndi.properties config file.
+  template:
+    src: "jndi.properties.j2"
+    dest: "/tmp/jndi.properties"
+    mode: 0744
```
Comment (sgahlot) [inline automation/ansible/roles/setup_jmeter_client/tasks/main.yml:51]: mode should be 644

File: automation/ansible/variable_override.yml [hunk 4]
```diff
@@ -41,3 +41,12 @@ r2_broker07_live_ip:      52.116.135.19
...
 r2_vpc2_nfs_server_ip:    52.116.202.69
 r2_router03_spoke_ip:     52.116.129.90
+
+
+## Jmeter client setup variables.
+jmeter_binary_download_uri: https://dlcdn.apache.org/jmeter/binaries
+jmeter_binary_version: apache-jmeter-5.4.3
+jmeter_binary_archive: "{{ jmeter_binary_version }}.zip"
+jmeter_remote_bin_dir: jmeter
+jmeter_hub_router_ip: 163.75.95.11
```
Comment (sgahlot) [inline automation/ansible/variable_override.yml:51]: You can simply use following to setup the private IP and it won't need to change at all:
`jmeter_hub_router_ip: "{{ r1_vpc1_private_ip_prefix }}.128.100"`

Also, the jmeter properties should go in the `hosts.yml` as they should have some default values.

**Review by sgahlot — COMMENTED** [review]
@lokeshrangineni I've added some inline comments for various files. Please review them when you get a chance.

File: automation/ansible/variable_override.yml (hunk 4 above)
Comment (lokeshrangineni) [reply automation/ansible/variable_override.yml:51]: Made the changes and I will be checking in soon.

File: automation/ansible/roles/setup_jmeter_client/tasks/main.yml (hunk 3 above)
Comment (lokeshrangineni) [reply automation/ansible/roles/setup_jmeter_client/tasks/main.yml:51]: Made the changes and I will be checking in soon.

File: automation/ansible/roles/setup_jmeter_client/tasks/main.yml (hunk 2 above)
Comment (lokeshrangineni) [reply automation/ansible/roles/setup_jmeter_client/tasks/main.yml]: Made the changes and I will be checking in soon.

File: automation/ansible/roles/setup_jmeter_client/tasks/main.yml (hunk 1 above)
Comment (lokeshrangineni) [reply automation/ansible/roles/setup_jmeter_client/tasks/main.yml]: Made the changes and I will be checking in soon.

**Review by sgahlot — APPROVED** [review]
Looks good.

**Pull request merged**